### PR TITLE
Use a consistent process for the official release

### PR DESCRIPTION
**Pull request by smarterclayton**
All components are now built using hack/release.sh

[test]

### CODE REVIEW

File: hack/release.sh [hunk 1]
```diff
@@ -12,18 +12,16 @@ source "${OS_ROOT}/hack/lib/init.sh"
...
-fi
-tag="${OS_TAG}"
-
-if [[ "$(git name-rev --name-only --tags HEAD)" != "${tag}^0" ]]; then
-  if git rev-parse -q --short "${tag}" &>/dev/null; then
-    echo "Tag ${tag} already exists"
+tag="${OS_TAG:-}"
+if [[ -z "${tag}" ]]; then
+  if [[ "$( git tag --points-at HEAD | wc -l )" -ne 1 ]]; then
+    echo "error: Specify OS_TAG or ensure the current git HEAD is tagged."
```
Comment (stevekuznetsov) [inline hack/release.sh]: Since you've got them imported... `os::log::error`?

File: hack/release.sh [hunk 2]
```diff
@@ -12,18 +12,16 @@ source "${OS_ROOT}/hack/lib/init.sh"
...
     exit 1
-  else
-    git tag "${tag}" -a -m "${tag}" HEAD
+  fi
+  tag="$( git tag --points-at HEAD )"
+else
+  if [[ "$( git rev-parse "${tag}" )" != "$( git rev-parse HEAD )" ]]; then
+    echo "warning: You are running a version of hack/release.sh that does not match OS_TAG - images may not be build correctly"
```
Comment (stevekuznetsov) [inline hack/release.sh]: `os::log::warning`?

File: hack/release.sh [hunk 3]
```diff
@@ -12,18 +12,16 @@ source "${OS_ROOT}/hack/lib/init.sh"
...
+tag="${OS_TAG:-}"
+if [[ -z "${tag}" ]]; then
+  if [[ "$( git tag --points-at HEAD | wc -l )" -ne 1 ]]; then
+    echo "error: Specify OS_TAG or ensure the current git HEAD is tagged."
     exit 1
-  else
-    git tag "${tag}" -a -m "${tag}" HEAD
+  fi
+  tag="$( git tag --points-at HEAD )"
+else
```
Comment (stevekuznetsov) [inline hack/release.sh]: Can collapse into `elif`

**Comment by smarterclayton** [comment]
[test] #9203 test_pull_requests_origin_integration
<https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin_integration/4725/>

**Comment by smarterclayton** [comment]
[test]

**Comment by smarterclayton** [comment]
[merge]

**Comment by smarterclayton** [comment]
[merge] #10274
https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin_check/4736/

**Pull request merged**